### PR TITLE
engineering: debugging in JavaScript

### PR DESCRIPTION
**Pull request by namtrhg**
#### What's this PR do?

- [x] Add some notes on using `console.log`, `console.assert`, `console.dir`, `console.time` & `console.timeEnd` for better debugging experience in JavaScript development.
- [x] Add some notes on debugging using the `debugger` statement.

### CODE REVIEW

File: Engineering/Debugging in JavaScript.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+```
+Another useful difference in Chrome exists when sending **`DOM`** elements to the console:
+![console.dir() example](https://pbs.twimg.com/media/Di30ycpW0AAYOx7.jpg:medium)
+
+### `console.time()` and `console.timeEnd()`
+The `console.time()` method launches a timer that you may use to track the duration of the operation. You may have up to 10,000 timers running on a single page, giving each one a unique name. When you use `console.timeEnd()` with the same name, the browser returns the time in milliseconds since the timer was started.
+
+#### Syntax
+```js
+time(label)
```
Comment (tienan92it) [inline Engineering/Debugging in JavaScript.md:88]: does this mean `console.time(label)`?

Comment (tienan92it) [reply Engineering/Debugging in JavaScript.md:88]: should add `console.timeEnd(label)` too

Comment (namtrhg) [reply Engineering/Debugging in JavaScript.md:88]: It is just a label to give the timer a name and this is optional.

Comment (tienan92it) [reply Engineering/Debugging in JavaScript.md:88]: I mean line 88 syntax should be `console.time(label)`, right?

**Pull request merged**